src/librustc_typeck/check/method/suggest.rs

```diff
@@ -256,15 +265,19 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                         err.emit();
                         return;
                     } else {
-                        struct_span_err!(
+                        let mut err = struct_span_err!(
                             tcx.sess,
                             span,
                             E0599,
                             "no {} named `{}` found for type `{}` in the current scope",
                             type_str,
                             item_name,
                             ty_string
-                        )
+                        );
+                        if let Some(suggestion) = suggestion {
+                            err.note(&format!("did you mean `{}::{}`?", type_str, suggestion));
```

Comment:
Can you make this a span_suggestion call?

Reply:
I did at first, the rendering is ugly because we use the same span.

Reply:
`span_suggestion_short` helps in many such cases, maybe here, too?